src/lib.rs

```diff
@@ -354,6 +360,67 @@ impl EvmContract {
         let addr = utils::hex_to_evm_address(&address);
         utils::u256_to_balance(&self.nonce_of(&addr))
     }
+
+
+    /// Register an ECDSA alias. This is an ECDSA key that is allowed to make updates and sign
+    /// messages on behalf of your Near-EVM account. This allows Near-EVM contracts to effectively
+    /// use EcRecover to seamlessly use Ethereum metatransactions and other signature-driven
+    /// features without integrating Near's signature scheme into the Near-EVM.
+    ///
+    /// To register an alias, the key must sign the Near-EVM address, AND the Near account
+    /// corresponding to that address must submit the signature. This prevents replay attacks,
+    /// and registering keys that are not actually controlled by the Near account holder.
+    ///
+    /// The message format uses the standard `Ethereum Signed Message` format, and can be produced
+    /// by `eth_sign` or `eth_signPersonal`. The body of the message is the account ID prepended
+    /// by the string "Near ecrecover alias: ". Note the lack of new line.
+    ///
+    /// # Note
+    ///
+    /// The `signature` argument is a hex string encoding the sigature in RSV format. This is the
+    /// typical output of a call to `personal_sign` in Metamask. It is usually exactly 65 bytes
+    /// long (130 hex characters), but may be longer if the chain ID is set to a high value. The
+    /// encoded string is NOT prepended by "0x".
+    ///
+    /// This contrasts with the VRS representation used by the EVM internally, and the two should
+    /// not be confused.
+    ///
+    /// # Arguments
+    ///
+    /// * `signature` - a valid ethereum-formatted signature
+    ///
+    /// # Panics
+    ///
+    /// * When `signature` is not valid hex.
+    /// * When the signature is invalid.
+    pub fn register_ecdsa_alias(&mut self, signature: String) {
```

Comment:
can an address register multiple aliases?

Reply:
yes, but there's currently no way to unregister them.

Reply:
Also this may have to have a return value so web3 doesn't mistake it for a revert